include/real/real.hpp

```diff
@@ -547,6 +547,21 @@ namespace boost {
                 return recurse_op(other, rc_lvl, OPERATION::SUBTRACTION);
             }
 
+
+
+            /*      POWER METHOD
+             *  @brief:  Calculates real_num^exponent
+             *  @params: real_num: boost real number whose power is to be evaluated
+             *  @params: exponent: power to which real_num is to be raised (needs to be an integer)
+             *  @return: returns a new boost real whose value is real_num^exponent
+             *  @author: Kishan Shukla
+             */
+
+            static real power(real<T> real_num, real<T> exponent){
+
+                return real(real_operation<T>(real_num._real_p, exponent._real_p, OPERATION::POWER));
+            }
+
```

Comment:
Yes

Reply:
First. For integer it must not be available as we have decided to let integer as a backend data type. 

Second, all real numbers should implement the interface of a real number, and constructions should be in a modular fashion where you construct the power function to work with any real number that implements the real numbers interface (basically a precision iterator that iterates the boundaries of its converging intervals). Then, the power function uses those intervals to oreturn the result. If you agree with this, then the rational number should have a precision iterator implemented instead of implementing the power for the rational data type.

Reply:
Here, by integers I meant integer type rational numbers. Sorry for not writing it clearly.

Now about precision iterator for real_rational number. I think the main idea behind implementing integers(integer type rational number) and rational number types were to avoid unnecessary iterations. If we make precision iterators for these numbers too, their implementation will be completely redundant, I mean then a rational number is simply equivalent to division(real_operation) between two numbers(real_explicit), the whole point of having integers and rational numbers will be gone.
As previously implemented any operation between two real_rational numbers is a real_rational number, then same should be true for powers also. 
And, if someone needs to use precision iterators for real_rational numbers, then I have made a functions "to_explicit" and "to_operation" which will convert an integer type (real_rational) number to explicit number and a rational number to real_operation number.
I suggest we should not have iterators for real_rational numbers, instead, we should return real_rational for power of real_rational, and if someone wants precision iterator, then we have those function "to_explicit" and "to_operation".